### PR TITLE
[harfbuzz,skia] Update and replace Skia dependencies with vcpkg

### LINKED ISSUE #10075: [harfbuzz] build failure
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4205960/install-x86-windows-dbg-out.log)
**Host Environment**
- OS: Windows 7
- Compiler: 19.00.24215.1 for x86

**To Reproduce**
`vcpkg.exe install harfbuzz[icu] --triplet x86-windows --recurse`

**Failure logs** 
```
Computing installation plan...
The following packages will be built and installed:
    harfbuzz[core,icu,ucdn]:x86-windows
Starting package 1/1: harfbuzz:x86-windows
Building package harfbuzz[core,icu,ucdn]:x86-windows...
Warning: abi keys are missing values:
    freetype
    ragel

-- Downloading https://github.com/harfbuzz/harfbuzz/archive/2.5.3.tar.gz...
-- Extracting source C:/work/vcpkg/downloads/harfbuzz-harfbuzz-2.5.3.tar.gz
-- Applying patch 0001-fix-cmake-export.patch
-- Applying patch 0002-fix-uwp-build.patch
-- Applying patch 0003-remove-broken-test.patch
-- Applying patch find-package-freetype-2.patch
-- Applying patch glib-cmake.patch
-- Using source at C:/work/vcpkg/buildtrees/harfbuzz/src/2.5.3-984271955b
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: C:/work/vcpkg/downloads/tools/cmake-3.14.0-windows/cmake-3.14.0-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory: C:/work/vcpkg/buildtrees/harfbuzz/x86-windows-dbg
    See logs for more information:
      C:\libgdiplus\libgdiplus_cario\vcpkg\buildtrees\harfbuzz\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/harfbuzz/portfile.cmake:72 (vcpkg_install_cmake)
  scripts/ports.cmake:90 (include)
```
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/4205944/config-x86-windows-out.log)
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4205971/install-x86-windows-dbg-out.log)

### LINKED ISSUE #10895: harfbuzz update to 2.6.5
Library name: harfbuzz

New version number: 2.6.5

Other information that may be useful (release notes, etc...)

Hi! Please update to 2.6.5 and ideally if possible try to use the meson support as we are considering to replace our cmake and autotools support with meson in future so if is possible let us know issues you hit, either here or on https://github.com/harfbuzz/harfbuzz so we fix them upstream for the next harfbuzz release. I've done harfbuzz vcpkg version update twice in the past but a little tedious for me to setup vcpkg right now but still like to keep it going :) Thanks

### LINKED ISSUE #11285: [skia] build failure
**Host Environment**
- OS: Windows 10
- Compiler: Visual Studio 2019

**To Reproduce**
`./vcpkg install skia`

**Failure logs** 
```bash
PS D:\vcpkg> .\vcpkg.exe install skia
Computing installation plan...
The following packages will be built and installed:
    skia[core]:x64-windows
Starting package 1/1: skia:x64-windows
Building package skia[core]:x64-windows...
-- Using cached D:/vcpkg/downloads/skia-05676f7bc238f667de848dfd37b4aa3c01b69efb.tar.gz
-- Using source at D:/vcpkg/buildtrees/skia/src/3c01b69efb-0a98da8354
-- Syncing git dependencies...
-- Generating build (debug)...
-- Generating build (release)...
-- Building Skia (debug)...
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -C D:/vcpkg/buildtrees/skia/x64-windows-dbg skia
    Working Directory: D:/vcpkg/buildtrees/skia/src/3c01b69efb-0a98da8354
    See logs for more information:
      D:\vcpkg\buildtrees\skia\build-x64-windows-dbg-out.log
      D:\vcpkg\buildtrees\skia\build-x64-windows-dbg-err.log

Call Stack (most recent call first):
  ports/skia/portfile.cmake:107 (vcpkg_execute_build_process)
  scripts/ports.cmake:90 (include)


Error: Building package skia:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: skia:x64-windows
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above.
```

**Additional context**

**build-x64-windows-dbg-err.log**
```bash
ninja: error: '../src/3c01b69efb-0a98da8354/third_party/externals/icu/common/icudtl.dat', needed by 'icudtl.dat', missing and no known rule to make it
```
I really don't have this file here
图片

### PR DESCRIPTION
**Pull request by vejmartin**
Updates HarfBuzz to the latest release (2.6.6) and replaces most in-source Skia dependencies with ones provided by vcpkg.

Fixes #11285, fixes #10895, fixes #10075

- Which triplets are supported/not supported? 
N/A

- Have you updated the CI baseline?
N/A

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/harfbuzz/portfile.cmake [hunk 1]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (LilyWangL) [inline ports/harfbuzz/portfile.cmake:3]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/harfbuzz/portfile.cmake [hunk 2]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO harfbuzz/harfbuzz
-    REF 2.5.3
-    SHA512 d541463b3647fc2c7ddaa29aedcea1c3bde5e26e0d529384d66d630af3aaf2a4befb3c4d47c93833f099339a0f951fb132011a02c57fc00ba543bd1b17026ffa
+    REF 2.6.6
+    SHA512 3ddf3e6eccf28ca1441544f0b67e243c6a85a32122bfc0f8092b3cc465b20a25aa3cb72404070d2627b9e204f86412c3bfb9aaca272c5492d8448facc1971a7d
     HEAD_REF master
     PATCHES
         0001-fix-cmake-export.patch
```
Comment (LilyWangL) [inline ports/harfbuzz/portfile.cmake:10]: Please fix the following comments:
1. Please use [vcpkg_check_features()](https://github.com/microsoft/vcpkg/blob/a3a6530631df905eb5c0e26d0b20d7d548e0c465/docs/maintainers/vcpkg_check_features.md) to handle features.
2. Please use `file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright.
3. `vcpkg_test_cmake` is deprecated. Please remove it.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by vejmartin** [comment]
@LilyWangL, thank you for linking the HarfBuzz PRs! I've added the changes from https://github.com/microsoft/vcpkg/pull/10951.

The Windows CIs were failing with a `[System.IO.Directory]::Delete, The directory specified by path is not empty.` error, so I also made a change in `scripts/azure-pipelines/windows/initialize-environment.ps1`.

**Review by LilyWangL — APPROVED** [review]

**Comment by strega-nil** [comment]
This looks great! Thanks so much @vejmartin :)

**Comment by ebraminio** [comment]
Hey there, this looks great! I was requesting meson integration also on #10895 for harfbuzz, can you have a look and maybe bring the needed patches to harfbuzz upstream? Thanks!

**Comment by vejmartin** [comment]
Hey @ebraminio, yes, I'm happy to take a look :)

**Comment by ebraminio** [comment]
@vejmartin Hey there, any chance to have a look at it again? I've also tried that https://github.com/microsoft/vcpkg/commit/681a100df21ad6bcf909894860b8c40ffcc6f6cc the problem is `vcpkg_check_features` should give enabled/disabled instead on/off for meson and there are some issue on install parts as lack of `vcpkg_fixup_cmake_targets` equivalent for meson, otherwise it is working locally at least as far as I can say. Any chance you like to have a look at it? We have a user that https://github.com/harfbuzz/harfbuzz/issues/2586#issuecomment-663880230 @wangqr that likes to use vcpkg and drop direct cmake use if this could happen so it really helps upstream on moving the project forward also. If you can't of course please let me know so I can see what I can do myself. Filed https://github.com/microsoft/vcpkg/issues/12572 also. Thank you very much

**Pull request merged**